File: Extras Built In function 2/Temperature.cs
using System;

class Temperature
{
    static double CelsiusToFahrenheit(double celsius)
    {
        return (celsius * 9 / 5) + 32;
    }

    static double FahrenheitToCelsius(double fahrenheit)
    {
        return (fahrenheit - 32) * 5 / 9;
    }

    static void Main()
    {
        Console.Write("Enter temperature: ");
        double temp = double.Parse(Console.ReadLine());

        Console.Write("Convert to (C/F)? ");
        char choice = char.ToUpper(Console.ReadKey().KeyChar);
        Console.WriteLine();

        if (choice == 'C')
            Console.WriteLine($"Temperature in Celsius: {FahrenheitToCelsius(temp)}");
        else if (choice == 'F')
            Console.WriteLine($"Temperature in Fahrenheit: {CelsiusToFahrenheit(temp)}");
        else
            Console.WriteLine("Invalid choice.");
    }
}

File: NodeJSProgram/Functions/Temperature.js
function convertTemperature(choice, temp) {
    switch (choice) {
        case "1": // Celsius to Fahrenheit
            if (temp >= 0 && temp <= 100) {
                return `${temp}°C = ${(temp * 9 / 5 + 32).toFixed(2)}°F`;
            } else {
                return "Temperature out of range (0°C - 100°C)";
            }
        case "2": // Fahrenheit to Celsius
            if (temp >= 32 && temp <= 212) {
                return `${temp}°F = ${((temp - 32) * 5 / 9).toFixed(2)}°C`;
            } else {
                return "Temperature out of range (32°F - 212°F)";
            }
        default:
            return "Invalid Choice";
    }
}